src/build/build_step.go

```diff
@@ -191,9 +198,11 @@ func buildTarget(tid int, state *core.BuildState, target *core.BuildTarget) (err
 		// Note that ordering here is quite sensitive since the post-build function can modify
 		// what we would retrieve from the cache.
 		if target.PostBuildFunction != nil && !haveRunPostBuildFunction {
-			log.Debug("Checking for post-build output file for %s in cache...", target.Label)
-			if state.Cache.RetrieveExtra(target, cacheKey, target.PostBuildOutputFileName()) {
-				if postBuildOutput, err = runPostBuildFunctionIfNeeded(tid, state, target, postBuildOutput); err != nil {
+			log.Debug("Checking for post-build output for %s in cache...", target.Label)
+			if metadata := state.Cache.Retrieve(target, cacheKey); metadata != nil {
+				storePostBuildOutput(target, metadata.Stdout)
+				postBuildOutput = string(metadata.Stdout)
+				if err := runPostBuildFunctionIfNeeded(tid, state, target, postBuildOutput); err != nil {
```

Comment:
Given we know that PostBuildFunction is not nil here (line 193) can we just inline the call here? (See also line 125 for identical concern)

Reply:
yeah I think you're right